code-challenges.test.js

```diff
@@ -15,40 +15,93 @@
 
 // a) Create a test with expect statements using the variables provided.
 
-const secretCodeWord1 = "Lackadaisical"
-// Expected output: "L4ck4d41s1c4l"
-const secretCodeWord2 = "Gobbledygook"
-// Expected output: "G0bbl3dyg00k"
-const secretCodeWord3 = "Eccentric"
-// Expected output: "3cc3ntr1c"
-
-
+describe("code", () => {
+  it("converts 'a' to 4, 'e' to 3, 'i' to 1, and 'o' to 0", () => {
+    const secretCodeWord1 = "Lackadaisical"
+    const secretCodeWord2 = "Gobbledygook"
+    const secretCodeWord3 = "Eccentric"
+    expect(code(secretCodeWord1)).toEqual("L4ck4d41s1c4l")
+    expect(code(secretCodeWord2)).toEqual("G0bbl3dyg00k")
+    expect(code(secretCodeWord3)).toEqual("3cc3ntr1c")
+  })
+})
 
 // b) Create the function that makes the test pass.
 
-
+//create result array
+//split input string into an array of letters and iterate over that array
+//lowercase character before checking if it is equal to lowercase input character
+//if the letter is one in the code, change it to the corresponding number
+//push each character into the result array
+//return result array joined on an emptry string
+
+const code = (str) => {
+ let result = []
+ let lettersArr = str.split('')
+ for (let i = 0; i < lettersArr.length; i++) {
+   let currentLetter = lettersArr[i]
+   let currentLowercase = lettersArr[i].toLowerCase()
+   if (currentLowercase === 'a') {
+     currentLetter = 4
+   } else  if (currentLowercase === 'e') {
+      currentLetter = 3
+   } else  if (currentLowercase === 'i') {
+       currentLetter = 1
+   } else  if (currentLowercase === 'o') {
+        currentLetter = 0
+      }
+      result.push(currentLetter)
+ }
+ return result.join('')
+}
```

Comment:
Looks good. You could shorten up your code a bit by replacing your currentLetter reassignment with a push action.